### PR TITLE
fix: enable ltcg on windows

### PR DESCRIPTION
**Pull request by nornagon**
Ref https://github.com/electron/electron/issues/18670

### CODE REVIEW

File: tools/generate_config_gypi.py [hunk 1]
```diff
@@ -2,10 +2,16 @@
...
+  config = {
+    'variables': {
+      'built_with_electron': 1,
+    }
+  }
+  if target_os == 'win':
+    config['variables']['node_with_ltcg'] = 'true'
   with open(out, 'w') as f:
-    f.write("{'variables':{'built_with_electron': 1}}\n")
+    f.write(repr(config) + "\n")
```
Comment (refack) [inline tools/generate_config_gypi.py]: Personally I wouldn't use `repr`, but rather `json.dumps`...

Comment (nornagon) [reply tools/generate_config_gypi.py]: it's not json :)

Comment (refack) [reply tools/generate_config_gypi.py]: Right you are...
https://github.com/electron/node/blob/229bd3245b2f54c12ea9ad0abcadbc209f8023dc/configure.py#L1602-L1603
...
https://github.com/electron/node/blob/229bd3245b2f54c12ea9ad0abcadbc209f8023dc/tools/gyp/pylib/gyp/input.py#L240-L243

But still... IMHO `repr` is less optimal then other options, and `json` is the most normative format that's compatible with GYP[3] ¯\\_(ツ)_/¯

Comment (refack) [reply tools/generate_config_gypi.py]: P.S. https://github.com/nodejs/node/pull/25798

Comment (nornagon) [reply tools/generate_config_gypi.py]: oh, i didn't realise that gyp could read json! cool, will update.

**Comment by nornagon** [comment]
Apparently this doesn't actually fix the issue according to @miniak

**Pull request merged**